src/callbacks/liquidity/Ramses/RamsesV2DTL.sol

```diff
@@ -0,0 +1,239 @@
+///
+/// @dev        As a general rule, this callback contract does not retain balances of tokens between calls.
+///             Transfers are performed within the same function that requires the balance.
+contract RamsesV2DirectToLiquidity is BaseDirectToLiquidity {
+    // ========== ERRORS ========== //
+
+    error Callback_Params_PoolFeeNotEnabled();
+
+    // ========== STRUCTS ========== //
+
+    /// @notice     Parameters for the onSettle callback
+    /// @dev        This will be encoded in the `callbackData_` parameter
+    ///
+    /// @param      maxSlippage             The maximum slippage allowed when adding liquidity (in terms of `ONE_HUNDRED_PERCENT`)
+    /// @param      veRamTokenId            The token ID of the veRAM token to use for the position (optional)
+    struct OnSettleParams {
+        uint24 maxSlippage;
+        uint256 veRamTokenId;
+    }
+
+    // ========== STATE VARIABLES ========== //
+
+    /// @notice     The Ramses V2 factory
+    IRamsesV2Factory public ramsesV2Factory;
+
+    /// @notice     The Ramses V2 position manager
+    IRamsesV2PositionManager public ramsesV2PositionManager;
+
+    // ========== CONSTRUCTOR ========== //
+
+    constructor(
+        address auctionHouse_,
+        address ramsesV2Factory_,
+        address payable ramsesV2PositionManager_
+    ) BaseDirectToLiquidity(auctionHouse_) {
+        if (ramsesV2Factory_ == address(0)) {
+            revert Callback_Params_InvalidAddress();
+        }
+        ramsesV2Factory = IRamsesV2Factory(ramsesV2Factory_);
+
+        if (ramsesV2PositionManager_ == address(0)) {
+            revert Callback_Params_InvalidAddress();
+        }
+        ramsesV2PositionManager = IRamsesV2PositionManager(ramsesV2PositionManager_);
+    }
+
+    // ========== CALLBACK FUNCTIONS ========== //
+
+    /// @inheritdoc BaseDirectToLiquidity
+    /// @dev        This function performs the following:
+    ///             - Validates the input data
+    ///
+    ///             This function reverts if:
+    ///             - OnCreateParams.implParams.poolFee is not enabled
+    ///             - The pool for the token and fee combination already exists
+    function __onCreate(
+        uint96,
+        address,
+        address baseToken_,
+        address quoteToken_,
+        uint256,
+        bool,
+        bytes calldata callbackData_
+    ) internal virtual override {
+        OnCreateParams memory params = abi.decode(callbackData_, (OnCreateParams));
+        uint24 poolFee = abi.decode(params.implParams, (uint24));
+
+        // Validate the parameters
+        // Pool fee
+        // Fee not enabled
+        if (ramsesV2Factory.feeAmountTickSpacing(poolFee) == 0) {
+            revert Callback_Params_PoolFeeNotEnabled();
+        }
+
+        // Check that the pool does not exist
+        if (ramsesV2Factory.getPool(baseToken_, quoteToken_, poolFee) != address(0)) {
```

Comment:
might not be necessary similar to the check in the V1 contract

Reply:
Done